atomata/capstone_WebClient
# Parsing

### Summary 
This PR implements the tree parsing of the apparatus json and it's integration in the apparatus list and action list
### Details
-implements the tree parsing of the json to represent the nested structure of the apparatus and the properties of each node ( identifier, type , input values , etc) 
- used the new parsing method to find the assetbundle nodes of the appratus and to show the button for each assetbundle in the apparatus list
- used the new parsing method to find the actions of each assetbundle and to show them as button in the action list
### How to test?
- run the project ( make sure the correct json files are uploaded to cloud first)
- select an appratus from appratus selection menu
- you should see all the components of the apparartus ( nested assetbundles) in the appratus box 
- after selecting each appartus you should see the actions it supports in the action list
- you should be able to add or remove actions from action sequence box
### Checks 
- [ X] Tested Changes
